### DISCUSSION TITLE
To restore the Verge wallet on the IOS phone, is it 12 words or 18 words?

### DISCUSSION THREAD

**Discussion by automaticprofits** [vergecurrency/vIOS#191] [Q&A] [answered]
Hello,

To restore the Verge wallet on the IOS, is it 12 words or 18 words?

I ask because when I tried to restore the Verge wallet for iPhone, it 
only asked for 12 words instead of the 18 words that I have saved.

Thank you.

**Comment by SwenVanZanten** [comment, marked as answer]
The iOS restore key is 12 words, but it also needs the pass phrase you created when setting the wallet up.
Otherwise it will open a new wallet.

So your HD private key is made up of two parts:
* The mnemonic: 12 words
* The pass phrase: Which you created yourself during the initial setup of your wallet

The possible mnemonic words can be found here: https://github.com/vergecurrency/BitcoinKit/blob/50c2c09656f7d1b0c3f159fe1b2cfa66fd2e0c11/Sources/BitcoinKit/Core/WordList.swift#L31

If you have listed any other word, it is wrong.

When restoring your wallet which different credentials from the original setup the application will open up a new empty wallet.
In this case you can remove that wallet again and retry. The reason for this is because the server doesn't know what is wrong or right.

**Reply by NikitaSheglov01** [reply]
Does anyone know how to restore the verge wallet through the seed phrase? lost wallet.dat file, don't understand how to get wallet back.
